OOP_ComposerLessons/Lesson11/src/Student.php

```diff
@@ -0,0 +1,59 @@
+<?php
+namespace Lesson11;
+/**
+ * [Description Student]
+ */
+class Student
+{
+    private string $name;
+    private int $course;
+    /**
+     * @param string $name
+     * @param int $course
+     */
+    public function __construct(string $name) 
+    {
+        
+        $this->name = $name;
+        $this->course = 1;
+       
+    }
+    /**
+     * @return string
+     */
+    public function getName() : string 
+    {
+        return $this->name;
+    }
+    /**
+     * @return int
+     */
+    public function getCourse() : int 
+    {
+        return $this->course;
+    }
+    /**
+     * @return Student
+     */
+    public function transferToNextCourse() : Student
+    {
+        $newCourse = $this->course + 1;
+        if ($this->isCourseCorrect($newCourse)) {
```

Comment:
We don't do anything when course is wrong, because the only case when it is wrong when the course equal to 6. So, if the course of a student is wrong after transferring, we keep an old value.